plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotTestingUtilities.java

```diff
@@ -0,0 +1,376 @@
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.TreeItem;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.swt.finder.SWTBot;
+import org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException;
+import org.eclipse.swtbot.swt.finder.finders.UIThreadRunnable;
+import org.eclipse.swtbot.swt.finder.results.Result;
+import org.eclipse.swtbot.swt.finder.results.VoidResult;
+import org.eclipse.swtbot.swt.finder.results.WidgetResult;
+import org.eclipse.swtbot.swt.finder.waits.DefaultCondition;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotButton;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotCheckBox;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotShell;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTable;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTree;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.eclipse.swtbot.swt.finder.widgets.TimeoutException;
+
+/**
+ * Provides helper methods to aid in SWTBot testing.
+ */
+public class SwtBotTestingUtilities {
+
+  private static class TreeCollapsedCondition extends DefaultCondition {
+    private final TreeItem tree;
+
+    private TreeCollapsedCondition(TreeItem tree) {
+      this.tree = tree;
+    }
+
+    public String getFailureMessage() {
+      return "Could not collapse the tree of " + tree.getText();
+    }
+
+    public boolean test() throws Exception {
+      return !isTreeExpanded(tree);
+    }
+  }
+
+  private static class TreeExpandedCondition extends DefaultCondition {
+    private final TreeItem tree;
+
+    private TreeExpandedCondition(TreeItem tree) {
+      this.tree = tree;
+    }
+
+    public String getFailureMessage() {
+      return "Could not expand the tree of " + tree.getText();
+    }
+
+    public boolean test() throws Exception {
+      return isTreeExpanded(tree);
+    }
+  }
+
+  /**
+   * The delay to use between a simulated key/button press's down and up events.
+   */
+  public static final int EVENT_DOWN_UP_DELAY_MS = 100;
+
+  public static void clickButtonAndWaitForWindowChange(SWTBot bot, final SWTBotButton button) {
+    performAndWaitForWindowChange(bot, new Runnable() {
+      public void run() {
+        button.click();
+      }
+    });
+  }
+
+  public static void clickOnTableCellValue(SWTBotTable table, int col, String value) {
+    String column = table.columns().get(col);
+    for (int row = 0; row < table.rowCount(); row++) {
+      String cellValue = table.cell(row, column);
+      if (cellValue.equals(value)) {
+        table.click(row, col);
+        break;
+      }
+    }
+  }
+
+  /**
+   * Create a java project with the specified project name. This function opens up the Java
+   * Perspective.
+   * 
+   * @param bot The current SWTWorkbenchBot object
+   * @param projectName Name of java project to be created
+   */
+  public static void createJavaProject(SWTWorkbenchBot bot, String projectName) {
```

Comment:
`SwtBotProjectActions` contains the same or very similar method, can they be consolidated?